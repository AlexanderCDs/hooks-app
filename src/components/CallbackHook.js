/** 
 * @author: alexandercddev
 * @description: 
 * @date: 04/Octubre/2021
**/
import React, { useCallback, useState } from 'react'
import ShowIncrement from './ShowIncrement';

export const CallbackHook = () => {
    const [counter, setCounter] = useState( 10 );
    //const increment = () => setCounter( counter + 1 ); 

    const increment = useCallback ( (number) => {
        setCounter( c => c + number );
    }, [ setCounter ]);

    return (
        <div className="animate__animated animate__backInDown">
            <h1>useCallback Hook</h1>
            <h3> Counter: { counter } </h3>
            <hr />
            <ShowIncrement increment={increment} />
        </div>
    )
}

export default CallbackHook;
