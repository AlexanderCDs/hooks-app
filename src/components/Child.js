/** 
 * @author: alexandercddev
 * @description: 
 * @date: 04/Octubre/2021
**/
import React from 'react'

export const Child = React.memo(({ numero, incrementar }) => {

    console.log('  Me volv√≠ a generar :(  ');

    return (
        <button
            className="btn btn-primary mb-2"
            onClick={ () => incrementar( numero ) }
        >
            { numero }
        </button>
    )
})

export default Child;
