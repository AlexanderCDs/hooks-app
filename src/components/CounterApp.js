/** 
 * @author: alexandercddev
 * @description: 
 * @date: 30/Septiembre/2021
**/
import React, { useState } from 'react';
import '../assets/sass/CounterApp.scss';

export const CounterApp = () => {
    const [state, setState] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30, 
    });
    const {counter1, counter2} = state;
    const handleAdd = () => {
        setState( state => (
            {
                ...state,
                counter1: counter1 + 1, 
            }
        ))
    }

    return (
        <>
            <h1>Counter1: { counter1 }</h1>
            <h1>Counter2: { counter2 }</h1>
            <hr/>
            <div className="stats">
                <button 
                    className="btn btn-primary"
                    onClick={() => handleAdd('counter1')}
                >
                    + 1
                </button>  
            </div>
            <code>{JSON.stringify(state)}</code>
        </>
    )
};

export default CounterApp;