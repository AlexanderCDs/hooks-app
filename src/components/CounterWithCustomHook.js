/** 
 * @author: alexandercddev
 * @description: 
 * @date: 30/Septiembre/2021
**/
import React from 'react'; 
import { useCounter } from '../hooks/useCounter';

export const CounterWithCustomHook = () => { 

    const {state, increment, decrement, reset} = useCounter(100);

    return (
        <>
            <h1>Counter with custom hook</h1>
            <h1>Counter { state }</h1>
            <hr></hr>
            <div className="stats">
                <button 
                    className="btn btn-primary" 
                    onClick={increment}
                >
                    + 1
                </button>
                <button 
                    className="btn btn-primary" 
                    onClick={reset}
                >
                    Reset
                </button>
                <button 
                    className="btn btn-primary" 
                    onClick={decrement}
                >
                    - 1
                </button>   
            </div> 
        </>
    )
};

export default CounterWithCustomHook;