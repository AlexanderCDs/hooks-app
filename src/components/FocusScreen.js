/**
 * @author: alexandercd
 * @description: 
 * @date: 03/octubre/2021
 * **/
import React, { useRef } from 'react'

const FocusScreen = () => { 
    const inputRef = useRef(); 
    const handleClick = () => {
        inputRef.current.select()
    }
    return (
        <div className="animate__animated animate__backInDown">
            <h1>Focus Screen</h1>
             
            <hr></hr>

            <input 
                ref={inputRef}
                className="form-control"
                placeholder="Your name"
            />

            <button 
                className="btn btn-outline-primary mt-2"
                onClick={handleClick}
            >
                Focus
            </button>
        </div>
    )
}

export default FocusScreen
