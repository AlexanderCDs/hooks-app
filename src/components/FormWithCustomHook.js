/** 
 * @author: alexandercddev
 * @description: 
 * @date: 30/Septiembre/2021
**/
import React, {useEffect} from 'react';
import '../assets/sass/effects.scss'; 
import { useForm } from '../hooks/useForm';

export const FormWithCustomHook = () => {
    const [formValues, handleInputChange] = useForm({
        name: '',
        email: '',
        password: ''
    });
    const { name, email, password } = formValues;

    useEffect(() => {
        console.log('email change')
    }, [email])

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formValues)
    } 
    return (
        <form onSubmit={handleSubmit} className="animate__animated animate__backInDown">
            <h1>Form with custom hook </h1>
            <hr/>
            <div className="form-group">
                <input
                    type="text"
                    name="name"
                    className="form-control"
                    placeholder="Your name"
                    autoComplete="off"
                    value={name}
                    onChange={handleInputChange}
                />
            </div>
            <div className="form-group">
                <input
                    type="email"
                    name="email"
                    className="form-control"
                    placeholder="example@example.com"
                    autoComplete="off"
                    value={email}
                    onChange={handleInputChange}
                />
            </div> 
            <div className="form-group">
                <input
                    type="password"
                    name="password"
                    className="form-control"
                    placeholder="Password"
                    autoComplete="off"
                    value={password}
                    onChange={handleInputChange}
                />
            </div> 
            <button type="submit" className="btn btn-primary">
                    Guardar
            </button> 
        </form>
    )
};

export default FormWithCustomHook;