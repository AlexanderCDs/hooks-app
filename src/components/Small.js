/** 
 * @author: alexandercddev
 * @description: 
 * @date: 04/Octubre/2021
**/
import React, { memo } from 'react'

export const Small = memo(({value}) => {
    console.log('Me renderice de nuevo, me va dar ansiedad!!')
    return (
        <small>
            {value}
        </small>
    )
})
