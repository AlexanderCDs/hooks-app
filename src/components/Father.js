/** 
 * @author: alexandercddev
 * @description: 
 * @date: 04/Octubre/2021
**/
import React, { useCallback, useState } from 'react'
import Child from './Child' 

export const Father = () => {

    const numeros = [2, 4, 6, 8, 10];
    const [valor, setValor] = useState(0);

    //const incrementar = ( num ) => setValor( valor + num );

    const incrementar = useCallback( (number) => {
        setValor( v => v + number)
    }, [setValor]);

    return (
        <div className="animate__animated animate__backInDown">
            <h1>Padre</h1>
            <p> Total: { valor } </p>

            <hr />

            {
                numeros.map( n => (
                    <Child 
                        key={ n }
                        numero={ n }
                        incrementar={ incrementar }
                    />
                ))
            }
            {/* <Hijo /> */}
        </div>
    )
}

export default Father;