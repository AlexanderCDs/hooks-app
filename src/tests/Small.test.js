import React from 'react';
import { shallow } from 'enzyme';
import { Small } from '../components/Small';

describe('Test <Samll />', () => {
    test('should show success', () => {
        const wrappeer = shallow(<Small value="Hola" />)
        expect( wrappeer ).toMatchSnapshot();
    })
    
})
