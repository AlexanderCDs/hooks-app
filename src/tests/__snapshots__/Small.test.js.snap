// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Samll /> should show success 1`] = `
<small>
  Hola
</small>
`;
